sympy/sets/sets.py

```diff
@@ -1427,13 +1427,23 @@ def _contains(self, other):
         return And(*[set.contains(other) for set in self.args])
 
     def __iter__(self):
+        no_iter = True
         for s in self.args:
             if s.is_iterable:
+                no_iter = False
                 other_sets = set(self.args) - set((s,))
                 other = Intersection(other_sets, evaluate=False)
-                return (x for x in s if x in other)
+                for x in s:
+                    c = other.contains(x)
+                    if c == True:
```

Comment:
`c is True` and `c is False` are much faster.  `if bool(c):` or `if bool(c) is True` are probably better if you want to test equivalence.